src/proxy/admin_progress.go

```diff
@@ -0,0 +1,98 @@
+/*
+ * Radon
+ *
+ * Copyright 2018-2019 The Radon Authors.
+ * Code is licensed under the GPLv3.
+ *
+ */
+
+package proxy
+
+import (
+	"fmt"
+
+	"router"
+
+	"github.com/radondb/shift/shift"
+	shiftLog "github.com/radondb/shift/xlog"
+	querypb "github.com/xelabs/go-mysqlstack/sqlparser/depends/query"
+	"github.com/xelabs/go-mysqlstack/sqlparser/depends/sqltypes"
+	"github.com/xelabs/go-mysqlstack/xlog"
+)
+
+// Progress
+type Progress struct {
+	log    *xlog.Log
+	router *router.Router
+	db     string
+	table  string
+}
+
+// NewProgress -- creates new progress.
+func NewProgress(log *xlog.Log, router *router.Router, db string, table string) *Progress {
+	return &Progress{
+		log:    log,
+		router: router,
+		db:     db,
+		table:  table,
+	}
+}
+
+// CheckDBTable check the database and table.
+func (progress *Progress) CheckDBTable() error {
+	if tableCfg, err := progress.router.TableConfig(progress.db, progress.table); err != nil {
```

Comment:
The order needs to be reversed. firstly, checktable, then get tableconfig and check the shardtype.  
Or  remove the process `CheckTable`,because `router.TableConfig` already includes a check to see if the table exists.